### PR TITLE
Support _total and _count works for hibernate search

### PR DESCRIPTION
**Pull request by jaison-b**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java [hunk 1]
```diff
@@ -621,8 +621,6 @@ private IBundleProvider executeQuery(String theResourceType, SearchParameterMap
 				if (queryCount == null || queryCount > resources.size()) {
 					// No limit, last page or everything was fetched within the limit
 					bundleProvider.setSize(getTotalCount(queryCount, theParams.getOffset(), resources.size()));
-				} else {
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/search/SearchCoordinatorSvcImpl.java:624]: Was this redundant?  Any idea why it was here?

File: hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java [hunk 2]
```diff
@@ -1490,6 +1498,81 @@ private IIdType withObservationWithValueQuantity(double theValue) {
...
+			IIdType idCode2 = myTestDataBuilder.createObservation(asArray(myTestDataBuilder.withObservationCode("http://example.com/", "code-2")));
+			IIdType idCode3 = myTestDataBuilder.createObservation(asArray(myTestDataBuilder.withObservationCode("http://example.com/", "code-3")));
+
+			myCaptureQueriesListener.clear();
+			IBundleProvider resultBundle = myTestDaoSearch.searchForBundleProvider("Observation?code=code-1,code-2,code-3&_offset=1");
+			myCaptureQueriesListener.logSelectQueriesForCurrentThread();
+
+			assertThat(resultBundle.getAllResourceIds(), containsInAnyOrder(idCode2.getIdPart(), idCode3.getIdPart()));
+			// make also sure no extra SQL queries were executed
+			assertEquals(1, myCaptureQueriesListener.getSelectQueriesForCurrentThread().size(), "bundle was built with no sql");
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java:1545]: Can we turn on resource storage and change this count to 0?

Comment (jmarchionatto) [reply hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java:1545]: What is turning on resource storage?

Comment (jmarchionatto) [reply hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java]: Sure. Good idea. Done

**Review by michaelabuckley — APPROVED** [review]
Looks good.

**Pull request merged**